src/ContractError.ts

```diff
@@ -0,0 +1,7 @@
+export class ContractError extends Error {
+    constructor(msg: string) {
+        super(msg);
+
+        Object.setPrototypeOf(this, ContractError.prototype);
```

Comment:
I like it, would you mind adding a reference to the reason behind this line?
```suggestion
        // https://github.com/Microsoft/TypeScript-wiki/blob/main/Breaking-Changes.md#extending-built-ins-like-error-array-and-map-may-no-longer-work
        Object.setPrototypeOf(this, ContractError.prototype);
```